LayerClient/LayerClient.py

```diff
@@ -0,0 +1,519 @@
+        conversation we wish to send to
+        - `sender`: A `LayerClient.Sender` instance
+        - `message_parts`: An array of `LayerClient.MessagePart` objects
+        - `notification`: Optional `PushNotification` instance.
+
+        Return: A new Message instance, or None if we were passed invalid
+        (None) arguments.
+        """
+
+        if not conversation or not sender or not message_parts:
+            return None
+
+        request_data = {
+            'sender': sender.as_dict(),
+            'parts': [
+                part.as_dict() for part in message_parts
+            ],
+        }
+        if notification:
+            request_data['notification'] = notification.as_dict()
+
+        return Message.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(
+                    LAYER_URI_CONVERSATIONS,
+                    conversation.uuid(),
+                    LAYER_URI_MESSAGES,
+                ),
+                request_data,
+            )
+        )
+
+    def send_announcement(self, sender, recipients, message_parts,
+                          notification=None):
+        """
+        Send an announcement to a list of users.
+
+        Parameters:
+        - `sender`: A `LayerClient.Sender` instance. The sender must have a
+            name, as this endpoint cannot be used with a sender ID.
+        - 'recipients`: A list of strings, each of which is a recipient ID.
+        - `message_parts`: An array of `LayerClient.MessagePart` objects
+        - `notification`: Optional `PushNotification` instance.
+        """
+        request_data = {
+            'sender': {
+                'name': sender.name,
+            },
+            'parts': [
+                part.as_dict() for part in message_parts
+            ],
+            'recipients': recipients,
+        }
+        if notification:
+            request_data['notification'] = notification.as_dict()
+
+        return Announcement.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(LAYER_URI_ANNOUNCEMENTS),
+                request_data,
+            )
+        )
+
+
+class Announcement(BaseLayerResponse):
+    """
+    Contains the data returned from the API when sending an Announcement
+    """
+
+    def __init__(self, id, url, sent_at, recipients, sender, parts):
+        self.id = id
+        self.url = url
+        self.sent_at = sent_at
+        self.recipients = recipients
+        self.sender = sender
+        self.parts = parts
+
+    @staticmethod
+    def from_dict(dict_data):
+        return Announcement(
+            dict_data.get('id'),
+            dict_data.get('url'),
+            Announcement.parse_date(dict_data.get('sent_at')),
+            dict_data.get('recipients'),
+            Sender.from_dict(dict_data.get('sender')),
+            [
+                MessagePart.from_dict(part) for part in dict_data.get('parts')
```

Comment:
super small, but you should do `dict_data.get('parts', [])` so if there are no parts it won't break